core/src/main/java/org/elasticsearch/cluster/routing/RoutingTable.java

```diff
@@ -222,6 +224,37 @@ public GroupShardsIterator allAssignedShardsGrouped(String[] indices, boolean in
         return new GroupShardsIterator(set);
     }
 
+    public ShardsIterator allShardsExcludingRelocationTargets(String[] indices) {
```

Comment:
relocating targets are something we create on the fly - most of the time they are not needed and do not really exists in the routing table (but do in routing nodes!). I think we can just call this allShards..